salt/orchestrate/edx/backup.sls

```diff
@@ -102,6 +124,35 @@ execute_enabled_backup_scripts:
         - salt: deploy_backup_instance_to_{{ ENVIRONMENT }}
         - salt: mount_backup_drive
 
+unmount_backup_drive:
+  salt.function:
+    - tgt: 'G@roles:backups and G@environment:{{ ENVIRONMENT }}'
+    - tgt_type: compound
+    - name: state.single
+    - arg:
+        - mount.unmounted
+    - kwarg:
+        name: /backups
+        device: /dev/xvdb
+    - require:
+        - salt: execute_enabled_backup_scripts
+
+detach_backup_volume:
+  salt.function:
+    - tgt: 'G@roles:backups and G@environment:{{ ENVIRONMENT }}'
+    - tgt_type: compound
+    - name: saltutil.runner
+    - arg:
+        - cloud.action
+    - kwarg:
+        func: boto_ec2.detach_volume
+        kwargs:
+          tags: {{ backup_volume_name }}
+          instance_id: {{ instance_id }}
```

Comment:
Right, but because we are creating the instance as part of this run the `instance_id` will not be resolved because the`find_instances` call will be executed before the instance is created.